pkg/operator/controller/ingress/status.go

```diff
@@ -716,14 +715,8 @@ func checkDefaultCertificate(secret *corev1.Secret, domain string) error {
 		if err != nil {
 			continue
 		}
-		foundSAN := false
-		for i := range cert.DNSNames {
-			if cert.DNSNames[i] == domain {
-				foundSAN = true
-			}
-		}
-		if cert.Subject.CommonName == domain && !foundSAN {
-			return fmt.Errorf("certificate in secret %s/%s has legacy Common Name (CN) but has no Subject Alternative Name (SAN) for domain: %s", secret.Namespace, secret.Name, domain)
+		if cert.SignatureAlgorithm == x509.SHA1WithRSA || cert.SignatureAlgorithm == x509.ECDSAWithSHA1 {
+			return fmt.Errorf("certificate in secret %s/%s has weak SHA1 signature algorithm: %s", secret.Namespace, secret.Name, domain)
```

Comment:
Worth a link out to https://docs.openshift.com/container-platform/4.16/release_notes/ocp-4-16-release-notes.html#ocp-4-16-networking ?  And possibly extending that section to fix "HaProxy" -> "HAProxy" and link https://docs.openshift.com/container-platform/4.16/networking/routes/secured-routes.html ?